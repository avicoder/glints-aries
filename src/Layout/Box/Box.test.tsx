import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import { Box } from './index';

describe('<Box />', () => {
  it('should match snapshot when margin is set', () => {
    const { asFragment } = render(<Box m={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when all margin directions are set', () => {
    const { asFragment } = render(<Box mt={64} mr={64} mb={64} ml={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when padding is set', () => {
    const { asFragment } = render(<Box p={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when all padding directions are set', () => {
    const { asFragment } = render(<Box pt={64} pr={64} pb={64} pl={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when responsive margin is set', () => {
    const { asFragment } = render(<Box m={{ default: 32, desktopS: 64 }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when responsive padding is set', () => {
    const { asFragment } = render(<Box p={{ default: 32, desktopS: 64 }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
