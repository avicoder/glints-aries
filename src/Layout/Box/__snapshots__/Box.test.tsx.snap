// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should match snapshot when all margin directions are set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 ebktHH"
  />
</DocumentFragment>
`;

exports[`<Box /> should match snapshot when all padding directions are set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 jWilAT"
  />
</DocumentFragment>
`;

exports[`<Box /> should match snapshot when margin is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 cMvEap"
  />
</DocumentFragment>
`;

exports[`<Box /> should match snapshot when padding is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 hA-dxnS"
  />
</DocumentFragment>
`;

exports[`<Box /> should match snapshot when responsive margin is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 cYQtfv"
  />
</DocumentFragment>
`;

exports[`<Box /> should match snapshot when responsive padding is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 iVdNyz"
  />
</DocumentFragment>
`;
